pkgs/development/libraries/readline/8.2.nix

```diff
@@ -35,13 +35,15 @@ stdenv.mkDerivation rec {
      in
        import ./readline-8.2-patches.nix patch);
 
+  hardeningDisable = lib.optionals stdenv.hostPlatform.isCygwin [ "fortify" ];
+
   patches = lib.optionals (curses-library.pname == "ncurses") [
     ./link-against-ncurses.patch
   ] ++ [
     ./no-arch_only-8.2.patch
   ]
   ++ upstreamPatches
-  ++ lib.optionals stdenv.hostPlatform.isWindows [
+  ++ lib.optionals stdenv.hostPlatform.isMinGW [
```

Comment:
What's the exact reason for changing from checking for windows in general to mingw? Would `isWindows || isMinGW` work better?

Reply:
We don't want to apply these patches under Cygwin. Cywin provides a POSIX API under Windows, so we want to use the normal POSIX calls which readline would usually make.

Reply:
I guess this could be something like `(stdenv.hostPlatform.isWindows && !stdenv.hostPlatform.isUnix)` if we wanted to mark Cygwin as the technically‐correct Windows + Unix and confuse everyone for the rest of time?

Reply:
@emilazy yeah really good point, MSVC is on its way. I actually like the suggestion of `isWindows && !isUnix`.

Reply:
Just be prepared for the fact that there’s probably a lot of things in the tree that assume the two are disjoint. But it’d be great to deconfuse that along the way.